proc-macros/src/render_client.rs

```diff
@@ -100,8 +100,31 @@ impl RpcDescription {
 			let params = method.params.iter().map(|(param, _param_type)| {
 				quote! { #serde_json::to_value(&#param)? }
 			});
-			quote! {
-				Some(vec![ #(#params),* ].into())
+			match method.param_format {
+				ParamFormat::Map => {
+					// Extract parameter names.
+					let param_names = extract_param_names(&method.signature.sig);
+					// Combine parameter names and values into tuples.
+					let params = param_names.iter().zip(params).map(|pair| {
+						let key = pair.0;
+						let value = pair.1;
+						quote! { (#key, #value) }
+					});
+					let jsonrpsee = self.jsonrpsee_client_path.as_ref().unwrap();
```

Comment:
Could this use `jprs_client_item()` and do something like `let json_value = self.jprs_client_item(quote! { types::JsonValue });`? or even re-use the `serde_json` lvar from line 99?